packages/compose/src/hooks/use-focus-outside/test/index.js

```diff
@@ -130,11 +130,19 @@ describe( 'useFocusOutside', () => {
 		} );
 		await user.click( button );
 
-		// Simulate a blur event and the wrapper unmounting while the blur event
-		// handler is queued
-		button.blur();
+		// Click outside the wrapper to trigger a blur event and queue the callback
+		const outsideButton = screen.getByRole( 'button', {
+			name: 'Button outside the wrapper',
+		} );
+		await user.click( outsideButton );
+
+		// Immediately unmount the component while the blur event is queued
+		// The callback should still be called.
 		unmount();
 
-		expect( mockOnFocusOutside ).not.toHaveBeenCalled();
+		// Wait for the queued timeout to execute
+		await new Promise( ( resolve ) => setTimeout( resolve, 10 ) );
+
+		expect( mockOnFocusOutside ).toHaveBeenCalled();
```

Comment:
I'd discourage a real-time sleep like this, both because it's pretty fragile, and because it artificially prolongs the test runtime (although not by much in this case).

Could we do something where we set up the mock implementation to finish the test? Either using [the `done` callback](https://jestjs.io/docs/asynchronous#callbacks), or maybe something with [`Promise.withResolvers`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/withResolvers), e.g.

```ts
const { promise, resolve } = Promise.withResolvers();
mockOnFocusOutside.mockImplementation( resolve );
unmount();
await promise;
```

Reply:
The `withResolvers` is such a nice addition to the promises ⭐

Reply:
TIL about `withResolvers()`. Thanks!

Reply:
Looks like `Promise.withResolvers()` is only available in Node.js 22+, but the project requires Node.js >=20.10.0. This was leading to unit test failures on CI.

I've reverted to using the traditional manual deferred promise pattern for better compatibility.